### ISSUE TITLE
[Debugger]Avoid need to modify pom.xml to support DataSonnet language

### ISSUE THREAD

**Issue by apupier** [camel-tooling/camel-idea-plugin#646]
currently to evaluate expressions using DataSonnet in debugger, you need to add to the pom.xml:

```
    <dependency>
        <groupId>org.apache.camel.springboot</groupId>
        <artifactId>camel-datasonnet-starter</artifactId>
    </dependency>
    <dependency>
        <groupId>org.scala-lang</groupId>
        <artifactId>scala-library</artifactId>
        <version>2.13.3</version>
    </dependency>
```

it is modifying the user project. it would be nice if these dependencies can be injected automatically when needed

**Comment by davsclaus** [comment]
Yeah we should actually go the other way. That languages shown in that "list" is only languages that are available in the running Camel application.

The intention is not that you can run any languages "as you wish" but use the languages that your Camel application are using.

**Comment by javaduke** [comment]
I like this idea, but is there any way in the plugin or in Camel to obtain a list of all languages in the classpath?

**Comment by davsclaus** [comment]
We can see from the JARs on the classpath which ones are langauges and via the camel-catalog find out their names.

An alternative is via JMX but then its "only languages that are active in use", eg you may have camel-datasonnet on classpath but it has not been in use yet, then its not registered in camel contexxt.

**Comment by javaduke** [comment]
Ah, the resolving the new language is taken care of in the `ManagedBacklogDebugger`, it calls `camelContext.resolveLanguage()` before evaluating the expression. 
So by "via JMX" did you mean there's an existing functionality that I can use or that we need to implement a new call?
Perhaps add something like `getResolvedLanguages()` to the Camel context?

**Comment by davsclaus** [comment]
Yeah I am adding JMX operations to the CamelContextMBean
https://issues.apache.org/jira/browse/CAMEL-17383
